### PR TITLE
add spotlights method and Spotlights model

### PR DESCRIPTION
**Pull request by fluffetyfluff**
This also fixes the incorrectly documented Spotlight model.

### CODE REVIEW

**Comment by tybug** [comment]
(see https://github.com/ppy/osu-web/issues/7557 for osu-web side tracking of incorrect documentation)

File: ossapi/ossapiv2.py [hunk 1]
```diff
@@ -554,6 +554,12 @@ def ranking(self,
         return self._get(Rankings, f"/rankings/{mode.value}/{type_.value}",
             params=params)
 
+    @request
+    def spotlights(self) -> Spotlights:
```
Comment (tybug) [inline ossapi/ossapiv2.py:558]: missing an import of `Spotlights`?

**Review by tybug — CHANGES_REQUESTED** [review]

File: ossapi/models.py [hunk 2]
```diff
@@ -378,12 +378,16 @@ class Search:
...
     participant_count: Optional[int]
-    name: int
+    name: str
     start_date: Datetime
     type: str
 
+@dataclass
+class Spotlights:
+    spotlights: List[Spotlight]
+
```
Comment (tybug) [inline ossapi/models.py:390]: to be honest I'd probably do away with this model entirely and just instantiate `List[Spotlight]` - I know that's not what the osu-web docs say but doesnt really make much sense, they return lists of models in other places

Comment (fluffetyfluff) [reply ossapi/models.py:390]: Is the goal to maintain 1:1 consistency with the docs or what's most logical? I agree the `Spotlights` model doesn't really make much sense

Comment (tybug) [reply ossapi/models.py:390]: consistency would be ideal but at the end of the day we're an api wrapper being consumed by other applications which don't care about the details of the api. Part of our job is to do what makes the most sense for consumers, even if it means we have to do more work to make it match the api.

File: ossapi/ossapiv2.py [hunk 3]
```diff
@@ -554,6 +554,14 @@ def ranking(self,
         return self._get(Rankings, f"/rankings/{mode.value}/{type_.value}",
             params=params)
 
+    @request
+    def spotlights(self) -> List[Spotlight]:
+        """
+        https://osu.ppy.sh/docs/index.html#get-spotlights
+        This method returns a list of `Spotlight`s rather than a `Spotlights` model,
+        but since that's all the model contains it is functionally equivalent.
```
Comment (tybug) [inline ossapi/ossapiv2.py:562]: eh I don't think this is necessary, at the most I would leave an inline comment above `self._get` (but even then probably not), docstrings are intended to be useful for consumers more than for developers. Consumers don't care why it's a list of spotlights, they just want to know the return type.

**Review by tybug — APPROVED** [review]

**Pull request merged**